### PR TITLE
fixed the handling of function variables in a URL

### PR DESCRIPTION
**Pull request by czardoz**

### CODE REVIEW

File: lib/collection/url.js [hunk 1]
```diff
@@ -307,7 +307,7 @@ _.extend(Url, /** @lends Url */ {
         }
 
         // extract the query string
-        p.query = _.get(url.match(/^\?([^#$]+)/), '[1]');
+        p.query = _.get(url.match(/^\?([^#]+)/), '[1]');
```
Comment (czardoz) [inline lib/collection/url.js]: What was the `$` for? EOL?

**Comment by czardoz** [comment]
Do not merge yet, investigating why this is needed in the first place

File: lib/collection/url.js (hunk 1 above)
Comment (shamasis) [inline lib/collection/url.js]: Yes. But seems not so correct now that I see it. There is a + operator. Can we add the failing test case and then the passing one?

**Comment by czardoz** [comment]
Fixed everything up, with tests. There's no failing test, @shamasis because the failure cannot be reproduced anymore.

**Pull request merged**